components/Tree.go

```diff
@@ -557,3 +528,48 @@ func (tree *Tree) ExpandAll() {
 		return true
 	})
 }
+
+func (tree *Tree) InitializeNodes(dbName string) {
+	rootNode := tree.GetRoot()
+	if rootNode == nil {
+		panic("Internal Error: No tree root")
+	}
+
+	var databases []string
+
+	if dbName == "" {
+		dbs, err := tree.DBDriver.GetDatabases()
+		if err != nil {
+			panic(err.Error())
+		}
+		databases = dbs
+	} else {
+		databases = []string{dbName}
+	}
+
+	for _, database := range databases {
+		childNode := tview.NewTreeNode(database)
+		childNode.SetExpanded(false)
+		childNode.SetReference(database)
+		childNode.SetColor(app.Styles.PrimaryTextColor)
+		rootNode.AddChild(childNode)
+
+		go func(database string, node *tview.TreeNode) {
+			tables, err := tree.DBDriver.GetTables(database)
+			if err != nil {
+				logger.Error(err.Error(), nil)
+				return
+			}
+
+			tree.databasesToNodes(tables, node, true)
+			App.Draw()
+		}(database, childNode)
+	}
+}
+
+func (tree *Tree) Refresh(dbName string) {
+	rootNode := tree.GetRoot()
+	rootNode.ClearChildren()
+	// readd nodes
```

Comment:
```suggestion
	// read nodes
```

Or


```suggestion
	// re-add nodes
```

Using readd is ambiguous

Reply:
this is fixed now, thanks for pointing out